File: core/src/net/natruid/jungle/utils/skill/Effect.java
package net.natruid.jungle.utils.skill;

public class Effect {
    public static final Effect EMPTY = new Effect();
    public static final Effect[] EMPTY_ARRAY = new Effect[0];

    public final String type;
    public final ModdedValue amount;
    public final ModdedValue duration;
    public final ModdedValue chance;

    public Effect(String type, ModdedValue amount, ModdedValue duration, ModdedValue chance) {
        this.type = type;
        this.amount = amount;
        this.duration = duration;
        this.chance = chance;
    }

    public Effect() {
        this("", ModdedValue.ZERO, ModdedValue.ZERO, ModdedValue.ONE);
    }
}

File: core/src/net/natruid/jungle/utils/skill/Effect.kt
package net.natruid.jungle.utils.skill

class Effect(
    val type: String = "",
    val amount: ModdedValue = ModdedValue.ZERO,
    val duration: ModdedValue = ModdedValue.ZERO,
    val chance: ModdedValue = ModdedValue.ONE
) {
    companion object {
        val EMPTY = Effect()
        val EMPTY_ARRAY = Array(0) { EMPTY }
    }
}
